pkg/controller/workload/workload_hash.go

```diff
@@ -42,12 +78,20 @@ func (h *HashName) StrToNum(str string) uint32 {
 	hash.Reset()
 	hash.Write([]byte(str))
 
+	if num, exits := h.strToNum[str]; exits {
+		return num
+	}
+
 	// Using linear probing to solve hash conflicts
 	for num = hash.Sum32(); num < math.MaxUint32; num++ {
-		if h.numToStr[num] == "" {
+		// Create a new item if we find an empty slot
+		if _, exists := h.numToStr[num]; !exists {
 			h.numToStr[num] = str
-			break
-		} else if h.numToStr[num] == str {
+			h.strToNum[str] = num
+			// Create a new item here, should flush
+			if err := h.flush(); err != nil {
```

Comment:
Can we write delta data? Flush all seems very expensive

Reply:
Sure. What about appending?

Reply:
> maybe yaml is simpler
> 
> ```
> foo: 123
> bar: 124
> ```

right, that's a better idea.